File: app/src/main/java/com/senerunosoft/puantablosu/model/Game.java
package com.senerunosoft.puantablosu.model;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class Game {

    private String gameId;
    private String gameTitle;
    private List<Player> playerList;
    private List<Score> score;

    public Game() {
    }

    public Game(String gameTitle, List<Player> playerList) {
        this.gameId = UUID.randomUUID().toString();
        this.gameTitle = gameTitle;
        this.playerList = playerList;
        this.score = new ArrayList<>();
    }

    public String getGameId() {
        return gameId;
    }

    public void setGameId(String gameId) {
        this.gameId = gameId;
    }

    public String getGameTitle() {
        return gameTitle;
    }

    public void setGameTitle(String gameTitle) {
        this.gameTitle = gameTitle;
    }

    public List<Player> getPlayerList() {
        return playerList;
    }

    public void setPlayerList(List<Player> playerList) {
        this.playerList = playerList;
    }

    public List<Score> getScore() {
        return score;
    }

    public void setScore(List<Score> score) {
        this.score = score;
    }
}

File: app/src/main/java/com/senerunosoft/puantablosu/model/Game.kt
package com.senerunosoft.puantablosu.model

import com.senerunosoft.puantablosu.model.config.IConfig
import com.senerunosoft.puantablosu.model.enums.GameType
import java.util.UUID

data class Game(
    val gameId: String = UUID.randomUUID().toString(),
    var gameTitle: String = "",
    var playerList: List<Player> = ArrayList(),
    var score: MutableList<Score> = mutableListOf(),
    var gameType: GameType = GameType.GenelOyun,
    var config: IConfig? = null
) {
    // Empty constructor for Gson
    constructor() : this(UUID.randomUUID().toString(), "", ArrayList(), ArrayList(), GameType.GenelOyun, null)
    // Constructor matching the original Java constructor
    constructor(gameTitle: String, playerList: List<Player>, gameType: GameType, config: IConfig?) : this(
        UUID.randomUUID().toString(),
        gameTitle,
        playerList,
        ArrayList(),
        gameType,
        config
    )
}